### DISCUSSION TITLE
Robotic Arm Resources

### DISCUSSION THREAD

**Discussion by nbering** [Roger-random/Sawppy_Rover#24] [Ideas]
# Background

When I first pitched the idea of building a clone of this Rover design I found on the internet, my wife asked me what on Earth I would do with a little Mars Rover. I told her the chassis was just the beginning, and I'd use it to learn more about robotics. Add an arm, build out interesting control software solutions, etc.

So I've been doing a lot of research about how I would build a robotic arm for my family's Sawppy Rover clone (which my daughter lovingly named Star). *As an aside here, Star is pink and white because we built a robot about 2 years ago, and my older daughter came home from school one day and refused to work on it because "robots are for boys". So this rover build is as girl-friendly as I can get.*
<img src="https://user-images.githubusercontent.com/4421371/119218026-05d63e00-baac-11eb-8fd2-3bb949c47c02.png" height=150 />

# Sawppy Rovers with Arms

I know I'm not the only one working on this concept, and some have already achieved this goal.

- [MadeEngineer](https://twitter.com/MadeEngineer/) on Twitter has built out an arm using NEMA17 motors for the joints, and shared a [first test video](https://twitter.com/MadeEngineer/status/1344297958465015816) in December 2020
    <img src="https://pbs.twimg.com/media/ErqPS8WXcAADt22?format=jpg&name=large" alt="Sawppy Arm based on NEMA17 stepper motors" height="150" />
- [Laura McKeegan](https://www.youtube.com/channel/UC7MpBDc42kKvXbNq57nYqsA) on YouTube has shared some stunning videos of [CJ](https://youtu.be/YvjbK8DOYGQ), a rover that is almost barely recognizable as being Sawppy because the cosmetic panels and MMRTG housing on it makes it such a perfect match to Curiosity. But if you pay attention to the rocker/bogie suspension you'll see all the Sawppy designs there. [CJ's arm](https://youtu.be/3rer-UcFSx0) appears to be mostly cosmetic since the tool turret is all non-functional 3D prints to mimic the origin's design, but the joint structure is true to Curiosity and made with LX-16A motors, like the rest of Sawppy.
    <img src="https://user-images.githubusercontent.com/4421371/119217855-12a66200-baab-11eb-9008-a291353290bd.png" alt="CJ, a 1/4 scale Curiosity Rover" height="150"/>

I'm still probably going to start from scratch on this part, even if anyone made their plans available, because that was one of my initial goals.

# Research and Resources

Now I get to the meat and potatoes of why I wanted to make this post. During my hunt for details about the Curiosity rover's design, I first came up only with the [NASA summary](https://mars.nasa.gov/msl/spacecraft/rover/arm/) which is at the top of a Google search for "Curiosity Arm". Also up there in this search is a great summary of the [Mars Exploration Rover arm design](https://mars.nasa.gov/mer/mission/rover/arm/) from twin rovers Spirit and Opportunity.

There's also a few videos from NASA press releases where they test these arms. But they all lacked the further technical details of the arm's mechanics that I craved so much. I wanted to know:

- The long part that protrudes from the join... is that a motor?
- If the long part is a motor, what is that larger radius part right at the joint for?

And this great piece of [raw footage](https://www.youtube.com/watch?v=avvjiHQtHWI) from JPL where they're debugging the incomplete Curiosity arm. This one's title reminded me that, like the recent Perseverance rover was called Mars 2020 before the naming contest results changed the name, Curiosity's project name was different during development. Curiosity's working name was "Mars Science Laboratory", or MSL.

Mars Science Laboratory (MSL) gets _way_ more useful resources and research papers, design drafts, and reports if I search for that.

Just to start, [Mars Science Laboratory Robotic Arm](https://www.esmats.eu/esmatspapers/pastpapers/pdfs/2011/billing.pdf) is a very detailed breakdown of the design process and requirements put upon Curiosity's arm. There's this great article [IEEE Spectrum](https://spectrum.ieee.org/automaton/robotics/space-robots/robot-arm-mars-2020-rover) that includes an interview with one of the engineers on the Arm project, who also mentions that the Arm had very different requirements, but was largely based on the MER's arm (which is probably why they are so visually similar in diagrams that  I find it is easy to confuse them).

## Mars Science Laboratory Rover Arm (MSL RA)

In the first paper in the paragraph above, I found my answer to what the larger-radius structure on the joint was, but it still lead to more questions. It refers to it as a "flex spool" in this digram.

![curiosity arm rendering](https://user-images.githubusercontent.com/4421371/119218893-8d25b080-bab0-11eb-8a1f-f21f0af1f619.png)

I didn't quite understand what I was looking at in this diagram of the cabling system until I read the paper I'll mention next, but here's where the "flex spools" that I still didn't understand fall into the overall arm design. But I'll expand it here out-of-order from my investigation. It turns out the cabling for Curiosity's arm is all essentially Flexible Printed Circuit (FPC), like the flat brownish like you'd find for the connectors in small electronics. If it's the same material you find across the rest of the industry, this is basically a film of a flexible but very durable Polyamide, with a layer of copper film bonded to it, which can be etched in the same way that regular Printed Circuit Boards are made.

![curiosity arm cabling system](https://user-images.githubusercontent.com/4421371/119218955-dd047780-bab0-11eb-9b3c-35c3ab5f670f.png)

Using FPC for the cabling allowed the rover's designers to carefully lay out the 200+ signal lines needed to control the Rover's arm, along with all of the instruments on the turret, so that noisy lines remained away from more sensitive lines. The entire cabling system was also a single run of about 3 meters, with some lines terminating at the joints to control the actuators on the arm itself.

So these "flex spools" I found appear to be a way to wrap the cable into a loop so that it can make the turns at the joints without a lot of stress on the FPC. They also protect the cabling from the martian environment, which they were concerned might wear at the protective layer that covers the thin copper part of these circuits.

# Mars Exploration Rover Arm

As I mentioned, this design got inherited a lot of its architecture from the MER mission's arm. And as research material goes, I struck gold when I went looking for material that covered this "flex spool" part.

[The Mars Exploration Rover Instrument Positioning System](https://www-robotics.jpl.nasa.gov/publications/Eric_Baumgartner/IPS_IEEE_Aerospace_2005_final.pdf) covers the MER arm design in much greater mechanical detail. It also used this FPC cabling system, though MER's was about half the length of MSL's at about 1.3 meters.

![image](https://user-images.githubusercontent.com/4421371/119219452-ada33a00-bab3-11eb-9314-fca3797fa227.png)

But by far, the most interesting figure to me was this mechanical breakdown of the Elbow Joint on MER in Figure 5. It shows the precise layout of the motor, positioning encoder, gearbox, and so on.

![image](https://user-images.githubusercontent.com/4421371/119219481-d62b3400-bab3-11eb-9cca-eb1dab3ba4dd.png)

# Conclusion

I was so excited when I found this that I just had to share. So there it is! I hope this research helps anyone else interested in building an arm for their Sawppy Rover, like me. I'd be interested in seeing what everyone else has found on their journey to building onto Sawppy. What has worked for arm construction? What hasn't?

**Comment by Roger-random** [comment]
This is great research, thank you for writing it up and sharing it with the rest of us!

I hadn't know about this "flex spool" design but thanks to you I now know what to look for. Examining my collection of rover pictures, I realized we can see more of them on the camera mast such as in the lower-right corner of this picture.

![Curiosity rover camera mast](https://user-images.githubusercontent.com/8559196/119220184-167fb780-ba9e-11eb-8fe5-5d6ec735ea0f.jpg)
([Head of Mast on Mars Rover Curiosity](https://www.nasa.gov/mission_pages/msl/multimedia/pia15106-anno.html))

Interestingly, this spool is only partially enclosed by a cage instead of fully enclosed like those on the arm. There's probably a whole other paper floating around out there to explain why but I failed to find it.

**Reply by nbering** [reply]
Thanks! I was originally focused on learning as much as I could about the mechanical components of the arm, but it quickly becomes apparent much of what we *see* of the arm is designed around this electrical routing. The mechanics are almost simple by comparison.

Not that building an actuator that works at -135℃ is easy, which becomes apparent if you look into why the launch was delayed by 2 years from its original 2009 target.
https://llis.nasa.gov/lesson/11501
